File: QueueReversal/QueueReversal.java
/*
https://www.geeksforgeeks.org/problems/queue-reversal/1
*/

class Solution {
    public void reverseQueue(Queue<Integer> q) {
        Stack<Integer>stack=new Stack();
        while(!q.isEmpty()){
            stack.add(q.poll());
        }
        while(!stack.isEmpty()){
            q.add(stack.pop());
        }
    }
}

File: QueueReversal/QueueReversal.cpp
/*
https://www.geeksforgeeks.org/problems/queue-reversal/1
*/

class Solution {
  public:
    void reverseQueue(queue<int> &q) {
        stack<int>stk;
        while(!q.empty()){
            stk.push(q.front());
            q.pop();
        }
        while(!stk.empty()){
            q.push(stk.top());
            stk.pop();
        }
    }
};